src/Umbraco.Infrastructure/PropertyEditors/ImageCropperPropertyValueEditor.cs

```diff
@@ -173,7 +176,7 @@ internal class ImageCropperPropertyValueEditor : DataValueEditor // TODO: core v
         }
 
         // process the file
-        var filepath = editorJson == null ? null : ProcessFile(file, cuid, puid);
+        var filepath = editorJson == null ? null : ProcessFile(file, _mediaService.GetById(editorValue.ContentKey)!, puid);
```

Comment:
When `editorValue.ContentKey` is returning an `Guid.Empty`, this will return null.

And explode later 🙈

Reply:
This is not going to work, when you are upload a media item by drag and drag, the FilePath service is called before the media item is actually persisted, so in that case GetById will always return null. This actually triggered why I made to request/PR, since it prevents me from being able to figure out information about the media item, if only a Guid is passed to me through this interface.

Reply:
See https://github.com/umbraco/Umbraco-CMS/blob/1d14158d82ce5d9e3ed9766e8e11faddb455be19/src/Umbraco.Web.BackOffice/Controllers/MediaController.cs#L796

This creates an IMedia with the a random GUID.

https://github.com/umbraco/Umbraco-CMS/blob/1d14158d82ce5d9e3ed9766e8e11faddb455be19/src/Umbraco.Web.BackOffice/Controllers/MediaController.cs#L801C21-L801C21

Then calls SetValue (https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Core/Extensions/ContentExtensions.cs#L291), which will call (https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Core/Extensions/ContentExtensions.cs#L359). This in turn calls TryGetMediaPath (https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Core/Extensions/ContentExtensions.cs#L376) which does all the IMediaPathScheme stuff. When it will then do SetValue (and thus go through the DataValueEditor), .Save has not been called on the IMediaItem yet, because that happens only after the SetValue from above has completed (https://github.com/umbraco/Umbraco-CMS/blob/1d14158d82ce5d9e3ed9766e8e11faddb455be19/src/Umbraco.Web.BackOffice/Controllers/MediaController.cs#L805).

A fix for this would be to CreateMediaWithIdentity, but that would also cause empty media items to be created if the upload fails... So this whole path is pretty touchy :P